### DISCUSSION TITLE
Using ReactiveUI with MS DependencyInjection

### DISCUSSION THREAD

**Discussion by XDgl233** [AvaloniaUI/Avalonia#13404] [Q&A]
As the title suggests, I am aware that the DI behind ReactiveUI uses Splat, and it has a package for compatibility with MS DI. But I still struggle to find useful articles or projects to see how to implement MS DI with RxUI, or if using splat is fine, how should I use it with Avalonia App? Currently my App.axaml.cs looks like this:

```
host = Host.CreateDefaultBuilder()
           .ConfigureServices((_, services) =>
           {
               services.UseMicrosoftDependencyResolver();
               var resolver = Locator.CurrentMutable;
               resolver.InitializeSplat();
               resolver.InitializeReactiveUI();

               DI_ServiceReg(services);
           })
           .Build()

Container = host.Services;
Container.UseMicrosoftDependencyResolver();
```

So far, my application can use CommunityToolkit MVVM no problem, but anything related to RxUI doesn't work, and RxUI Command will give errors like _Call from invalid thread_.
Very appreciate your help, thx :)

**Comment by maxkatz6** [comment]
`.UseReactiveUI()` must be called on Avalonia AppBuilder. And since you use hos builder with another layer of reactive UI initialization, you need to keep an eye on it, so it won't re-override Avalonia ReactiveUI integration by initializing reactiveui twice.

**Reply by maxkatz6** [reply]
Btw, what's the point of integrating ReactiveUI with MS DI? ReactiveUI can use its own DI internally, and you are not really required to change that IIRC, since you still can you MS DI independently.

**Reply by XDgl233** [reply]
Thank you :) I copied that code from the discussion [here](https://github.com/AvaloniaUI/Avalonia/discussions/8588). And I am quite new to these stuff and don't know much about what's really going on...
I will take a look and try your suggestion :)
